contracts/StakingV2.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.7;
```

Comment:
Should we think about upgrading?

Reply:
This comes with some issues for the rest of the contracts but we can see if it makes sense to do it!